### PR TITLE
Metadata cache exiftool singleton

### PR DESCRIPTION
**Pull request by amaleki**
Fixed all the test errors.  Update unit tests to setup/teardown ExifTool singleton.

### CODE REVIEW

File: elodie/external/pyexiftool.py [hunk 1]
```diff
@@ -151,8 +151,16 @@ def format_error (result):
...
+    """Metaclass to use the singleton [anti-]pattern"""
+    instance = None
 
-class ExifTool(object):
+    def __call__(cls, *args, **kwargs):
+        if cls.instance is None:
+            cls.instance = super(Singleton, cls).__call__(*args, **kwargs)
+        return cls.instance
+
+class ExifTool(object, metaclass=Singleton):
```
Comment (jmathai) [inline elodie/external/pyexiftool.py]: This line causes an [error](https://travis-ci.org/jmathai/elodie/jobs/636196855#L1648) in the python2 unit tests.

```
SyntaxError (invalid syntax (pyexiftool.py, line 163)) ... ERROR
```

File: elodie/tests/plugins/googlephotos/googlephotos_test.py [hunk 2]
```diff
@@ -34,6 +37,16 @@
 sample_metadata = sample_photo.get_metadata()
```
Comment (jmathai) [inline elodie/tests/plugins/googlephotos/googlephotos_test.py]: Looks like [Exiftool isn't running](https://travis-ci.org/jmathai/elodie/jobs/636196856#L1850) when getting the metadata. This error is via python3.

```
  File "/home/travis/build/jmathai/elodie/elodie/tests/plugins/googlephotos/googlephotos_test.py", line 37, in <module>
    sample_metadata = sample_photo.get_metadata()
  File "/home/travis/build/jmathai/elodie/elodie/media/base.py", line 93, in get_metadata
    'date_taken': self.get_date_taken(),
  File "/home/travis/build/jmathai/elodie/elodie/media/photo.py", line 61, in get_date_taken
    exif = self.get_exiftool_attributes()
  File "/home/travis/build/jmathai/elodie/elodie/media/media.py", line 122, in get_exiftool_attributes
    exif_metadata = ExifTool().get_metadata(source)
  File "/home/travis/build/jmathai/elodie/elodie/external/pyexiftool.py", line 338, in get_metadata
    return self.execute_json(filename)[0]
  File "/home/travis/build/jmathai/elodie/elodie/external/pyexiftool.py", line 320, in execute_json
    return json.loads(self.execute(b"-j", *params).decode("utf-8"))
  File "/home/travis/build/jmathai/elodie/elodie/external/pyexiftool.py", line 283, in execute
    raise ValueError("ExifTool instance not running.")
ValueError: ExifTool instance not running.
```

File: elodie/external/pyexiftool.py (hunk 1 above)
Comment (amaleki) [reply elodie/external/pyexiftool.py]: Singleton is now Python 2 & 3 compatible.

File: elodie/tests/plugins/googlephotos/googlephotos_test.py (hunk 2 above)
Comment (amaleki) [reply elodie/tests/plugins/googlephotos/googlephotos_test.py]: Moved some of the global variables back into the test functions so ExifTool could be initialized by setup_module before tests run.

**Comment by jmathai** [comment]
@amaleki could you sign the CLA for this?

**Comment by amaleki** [comment]
CLA fixed.

**Comment by jmathai** [comment]
Thanks! Looks **great**. I made a couple very minor changes and submitted a PR against your branch. Can merge this PR once you merge mine.

**Comment by amaleki** [comment]
Thank you for the feedback.  Changes looked good.  Approved and Merged.

**Pull request merged**